debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleOffsetContext.java

```diff
@@ -235,12 +254,20 @@ public Loader(OracleConnectorConfig connectorConfig) {
 
         @Override
         public OffsetContext load(Map<String, ?> offset) {
-            LcrPosition lcrPosition = LcrPosition.valueOf((String) offset.get(SourceInfo.LCR_POSITION_KEY));
-            Long scn = lcrPosition != null ? lcrPosition.getScn() : (Long) offset.get(SourceInfo.SCN_KEY);
             boolean snapshot = Boolean.TRUE.equals(offset.get(SourceInfo.SNAPSHOT_KEY));
             boolean snapshotCompleted = Boolean.TRUE.equals(offset.get(SNAPSHOT_COMPLETED_KEY));
+            Long scn;
+            if (adapter == OracleConnectorConfig.ConnectorAdapter.LOG_MINER) {
```

Comment:
Can this and other instances depending on the connector adapter be extracted directly into a s separate connector adapter startegy?